src/box-container/box-container.styles.tsx

```diff
@@ -50,27 +49,40 @@ export const Header = styled.div`
     align-items: center;
     justify-content: space-between;
 
-    ${MediaQuery.MaxWidth.mobileM} {
-        padding: 1rem;
+    ${MediaQuery.MaxWidth.mobileL} {
+        padding: 1rem 1.25rem;
+        display: block;
     }
 `;
 
 export const LabelText = styled(Text.H3)`
     word-wrap: break-word;
-    margin-right: 2rem;
-    display: block;
+    display: flex;
     flex: 1;
+    align-items: center;
+    p:last-child {
+        margin-right: 1rem;
+    }
 
-    ${MediaQuery.MaxWidth.mobileM} {
+    ${MediaQuery.MaxWidth.mobileL} {
         ${TextStyleHelper.getTextStyle("BodySmall", "semibold")}
-        margin-right: 1rem;
+        margin-right: 0rem;
+        display: flex;
+        justify-content: space-between;
     }
 `;
 
 export const LabelIcon = styled.span<LabelIconStyleProps>`
     margin-left: 0.625rem;
     font-size: 1.5rem;
     vertical-align: middle;
+    display: flex;
+    align-items: center;
+    margin-right: 2.5rem;
```

Comment:
should this 2.5rem belong to the button instead?

for this layout,
```
| title | button | v |
```
the title and button will be close together